src/Iframe.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { createResource } from "simple-cache-provider";
+import { cache } from "./shared";
+
+function load(attributes) {
+  const { src, height, ...attrs } = attributes;
+  return new Promise((resolve, reject) => {
+    const iframe = document.createElement("iframe");
+    Object.keys(attrs).forEach(name => iframe.setAttribute(name, attrs[name]));
+    iframe.src = src;
+    iframe.onload = () => {
+      iframe.height = height;
+      resolve(iframe);
+    };
+    iframe.height = 0;
+    document.body.appendChild(iframe);
+  });
+}
+
+const resource = createResource(load, ({ href }) => href);
+
+export const Iframe = props => {
+  const iframeNode = resource.read(cache, props);
+  return <iframe ref={ref => ref.replaceWith(iframeNode)} {...props} />;
```

Comment:
We might need to use something other than `replaceWith`, there are [some gaps in browser support](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/replaceWith#Browser_compatibility).  We'll also need to handle the case where `ref` is null.

Reply:
Turns out this wasn't necessary.